gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/resources/handler/authorization/AuthorizationRequestFailureHandler.java

```diff
@@ -167,10 +168,10 @@ private void processOAuth2Exception(AuthorizationRequest authorizationRequest,
             authorizationRequest.setRedirectUri(defaultErrorURL);
         }
         // check if the redirect_uri request parameter is allowed
-        if (client != null
+        if ((client != null
                 && client.getRedirectUris() != null
                 && authorizationRequest.getRedirectUri() != null
-                && !client.getRedirectUris().contains(authorizationRequest.getRedirectUri())) {
+                && !client.getRedirectUris().contains(authorizationRequest.getRedirectUri())) || invalidParamRedirectURI(client, authorizationRequest.getRedirectUri())) {
```

Comment:
**nitpick**: maybe we can move the first condition in a method named `forbiddenRedirectUri` for readability :thinking: ?

` if (forbiddenRedirectUri(client, authorizationRequest) || invalidParamRedirectURI(client, authorizationRequest.getRedirectUri()))`

Reply:
done, refactored and moved to a method as suggested